docs/android/codestyle.md

```diff
@@ -74,3 +74,121 @@ TODOs in code tend to be forgotten over time. When someone read them later, they
 ```bash
 // TODO Missing feature (linked issue #404)
 ```
+
+## Constants
+
+### Naming conventions
+
+We follow the [Kotlin property naming guidelines](https://kotlinlang.org/docs/coding-conventions.html#property-names).
+
+### Avoid magic numbers and strings
+
+Magic numbers or strings in code can make it difficult to understand the purpose of a value, leading to poor maintainability. Always replace magic numbers or strings with named constants.
+
+#### ❌ Don't do this
+
+```kotlin
+if (value == 42) {
+  // Do something
+}
+```
+
+In this example, it’s unclear why the value 42 is being used. At the very least, you should add a comment explaining its purpose. Even better for future evolutions, it should be defined as a constant.
+
+#### ✅ Do this
+
+```kotlin
+// Explanation or link about why we picked 42
+const val SUPER_IMPORTANT_THRESHOLD = 42
+
+if (value == SUPER_IMPORTANT_THRESHOLD) {
+    // Do something
+}
+```
+
+### Organizing constants
+
+General rule is if the constant is exposed outside of the file it needs to be properly be identifiable when imported either by its name or by its parents name.
+
+#### Within a dedicated file
+
+If you need to group related constants together, create an `object` to namespace them. Place the constants in a file suffixed with `*Constants.kt`. This approach improves organization and avoids cluttering unrelated files.
+
+**Example:**
+
+```kotlin
+// filepath: NetworkConstants.kt
+package package io.homeassistant.companion.android.network
+
+object NetworkConstants {
+    val TIMEOUT = 30.seconds
```

Comment:
if its a constant, why isnt it a const?

Reply:
Because in Kotlin only primitives types and strings can be mark as `const`. Here it's a `Duration` object and unfortunately it's not possible to mark it as a `const` We had the same discussion https://github.com/home-assistant/android/pull/5197#discussion_r2034645206.